sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs

```diff
@@ -93,5 +95,29 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
             builder.Services.AddSingleton<ServiceBusClientFactory>();
             return builder;
         }
+
+        public static IWebJobsBuilder AddServiceBusScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
+        {
+            ServiceBusScalerProvider provider = null; // lazy init
```

Comment:
A cleaner way to do this can be seen in an example [here](https://github.com/Azure/azure-functions-host/blob/d207a4380e18aa79a7545762bd2f5107f8338893/src/WebJobs.Script/ScriptHostBuilderExtensions.cs#L275) where the Functions Host has a ScriptHost singleton registered for multiple different interface types.

Reply:
I am not sure how to implement this, note we need to create an instance as we are passing trigger data. Can you please give me an example?

Reply:
Something like this:

```csharp
services.AddSingleton<ServiceBusScalerProvider>(s => new ServiceBusScalerProvider(s, triggerMetadata));
services.AddSingleton<IScaleMonitorProvider>(s => s.GetService<ServiceBusScalerProvider>());
services.AddSingleton<ITargetScalerProvider>(s => s.GetService<ServiceBusScalerProvider>());
```

Reply:
This approach does work as `ServiceBusScalerProvider` will be the same for all the triggers - last added by `services.AddSingleton<ServiceBusScalerProvider>` will be used.